### PR TITLE
build: bump golang v1.22

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
**Problem:**
Bump golang v.122

**Solution:**
Bump golang v.122

**Related Issue:**
https://github.com/harvester/harvester/issues/6160

**Test plan:**
No Need

### CODE REVIEW

**Review by bk201 — APPROVED** [review]

**Review by tserong — APPROVED** [review]

**Comment by Vicente-Cheng** [comment]
It's very weird that the segfault when running `go generate`.

I can reproduce this on my local. Let me dig more with that.

**Comment by tserong** [comment]
Thanks @Vicente-Cheng - I just noticed the segfault after I approved this one, then re-ran the job in case it was transient, but I guess it's not :-/

**Comment by Vicente-Cheng** [comment]
Hi guys, please help to review it again.

I also bumped the Wrangler and K8s versions.
The null pointer issue can be resolved by bumping the controller-tool version to v0.15.0

**Review by bk201 — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -36,21 +38,22 @@ require (
 	github.com/harvester/go-common v0.0.0-20231214093547-3e3f7fdd879a
 	github.com/jaypipes/ghw v0.8.1-0.20210701154532-dd036bd38c40
 	github.com/kevinburke/ssh_config v1.2.0
-	github.com/longhorn/longhorn-manager v1.5.3
+	github.com/longhorn/longhorn-manager v1.5.1
```
Comment (tserong) [inline go.mod:41]: The longhorn version number has gone backwards. Does this matter? (also, it's still v1.5 series...?)

**Review by tserong — COMMENTED** [review]
Just one question :-)

File: go.mod (hunk 1 above)
Comment (Vicente-Cheng) [reply go.mod]: Nice catch. Somehow, it went backward; move back to the v1.5.5.

**Review by tserong — APPROVED** [review]

**Comment by Vicente-Cheng** [comment]
@mergifyio backport v0.7.x

**Pull request merged**